lib/WorseReflection/Bridge/TolerantParser/Reflection/ReflectionClass.php

```diff
@@ -299,6 +310,33 @@ public function isInstanceOf(ClassName $className): bool
             return true;
         }
 
+        // do not try and reflect the parents if we can locally see that it is
+        // an instance of the given class
+        $baseClause = $this->node->classBaseClause;
+        if ($baseClause instanceof ClassBaseClause) {
+            $name = $baseClause->baseClass;
+            if ($name instanceof QualifiedName) {
+                if ((string)$name->getResolvedName() === $className->__toString()) {
+                    return true;
+                }
+            }
+        }
+
+        // do not try and reflect the parents if we can locally see that it is
+        // an instance of the given class
+        $baseClause = $this->node->classInterfaceClause;
+        /** @phpstan-ignore-next-line */
+        if ($baseClause instanceof ClassInterfaceClause) {
+            foreach ($baseClause->interfaceNameList->getElements() as $element) {
+                if (!$element instanceof QualifiedName) {
+                    continue;
+                }
+                if ((string)$element->getResolvedName() === $className->__toString()) {
+                    return true;
+                }
+            }
+        }
```

Comment:
update the comments here. this was to allow `instanceof` to work even if the type cannot be "found", as a side-effect this actually improves performance by around 5% in general.